publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/helper/RabbitMqProperties.java

```diff
@@ -259,6 +268,53 @@ private void madatoryParametersCheck() {
             }
         }
     }
+
+    /****
+     * This method is used to check for checking exchange availability, if
+     * exchange is not available creates a new exchange based on isCreateExchangeIfNotExisting boolean property .
+     * @throws IOException
+     * @throws InvalidExchangeException
+     */
+    private void madatoryCreateExchangeParametersCheck()
+            throws IOException, TimeoutException, InvalidExchangeException {
+        final boolean exchangeAlreadyExist = hasExchange(exchangeName);
+        if (!exchangeAlreadyExist) {
+            Channel channel = giveMeRandomChannel();
+            if (isCreateExchangeIfNotExisting()) {
+                try {
+                    channel.exchangeDeclare(exchangeName, "topic", true);
+                } catch (final IOException e) {
+                    throw new IOException("Failed to declare exchange " + exchangeName, e);
+                } finally {
+                    channel.close();
+                }
+            } else {
+                log.error(exchangeName + PropertiesConfig.EXCHANGE_UNAVAILABLE);
+                channel.close();
+                throw new InvalidExchangeException("Invalid exchange is given please change createExchangeIfNotExisting properties and try again");
+            }
+        }
+    }
+
+    /****
+     * This method is used to check whether exchange exists or not return boolean value
```

Comment:
complete the java doc
When will we return true? - if exchange already exists or other wise.